diesel/src/query_source/joins.rs

```diff
@@ -251,6 +288,15 @@ impl<T: Table, Selection> AppendSelection<Selection> for T {
     }
 }
 
+// TODO: not sure if this is the best way, but the existing impls aren't sufficient
+impl<Left: Expression + Clone, Selection> AppendSelection<Selection> for Nullable<Left> {
```

Comment:
The existing `AppendSelection` impls don't allow a `Nullable` left, but is there another type that might be more preferable to implement this on?

Reply:
Adding that impl should be fine.